src/lib/modules/Patron/PatronBulkExtendLoans/PatronBulkExtendLoans.js

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Icon, List, Modal } from 'semantic-ui-react';
+
+export default class PatronBulkExtendLoans extends Component {
+  state = { open: false };
+
+  open = () => this.setState({ open: true });
+  close = () => this.setState({ open: false });
+
+  handleSubmitExtend = () => {
+    const { bulkLoanExtension, patronPid } = this.props;
+
+    bulkLoanExtension(patronPid);
+    this.close();
+  };
+
+  render() {
+    const { patronPid, bulkLoanExtension, isLoading, ...props } = this.props;
+    const { open } = this.state;
+    return (
+      <Modal
+        open={open}
+        onClose={this.close}
+        onOpen={this.open}
+        trigger={
+          <Button
+            labelPosition="left"
+            fluid
+            icon
+            primary
+            loading={isLoading}
+            {...props}
+          >
+            <Icon name="refresh" />
+            Extend all loans
+          </Button>
+        }
+      >
+        <Modal.Header>Confirm extension action</Modal.Header>
+        <Modal.Content>
+          <Modal.Description>
+            The action will extend only the loans:{' '}
+            <List bulleted>
+              <List.Item>
+                <b>with the upcoming date of return,</b>
```

Comment:
* Loans that will have to be returned in next `{config.loanWillExpireDays}` days.
* Loans that do not involve third party libraries (Interlibrary Loans).
* Loans for which the literature has not other waiting loan's request.

I am not particular happy with the last one :)